drivers/modem/quectel-bg9x.c

```diff
@@ -1029,6 +1029,21 @@ static int modem_setup(void)
 			     NULL, 0U, "AT+QIACT=1", &mdata.sem_response,
 			     MDM_CMD_TIMEOUT);
 
+	/* If the PDP context is already opened ; it returns an error. In this case, close the context and reopens it. */
```

Comment:
I'm not sure why the PDP context is enabled as we've basically just started up the modem. 

Can you explain how you got to this case ? May be there is some case which needs a more proper fix.

Reply:
In my case, this issue might be caused by the fact that my module's reset is an hardware one (at the moment).

When the module isn't stopped/reseted properly, it seems to keep it's internal values & parameters. According to the datasheet, the module has 3Mb of Flash and 3Mb of RAM. I'm not sure of where the parameters of the module are stored, but It could in the Flash memory. In this case, being a non-volatile memory, the PDP parameters can be stored, and the PDP context still considered as activated, leading to the issue raised an solved here.

This fix is helping me to debug without having to reset the module each time by just de-activating the PDP context and re-activating it. Even thought it add weights on the code, it can be useful to some people. 

I need to dig deeper into it. In the meantime, please keep the PR open.

I am also pinging @gpaquet85 for his experience on another board than mine.

Reply:
I also habe this issue in two cases: 
   - when I start from scracth
   - when I restart modem (after PWREN management)
I also think that PDP are saved in memory and this case shall be managed to avoid this.
Maybe something else to manage it. We will have a look on it
Don't hesitate if you see something else

Reply:
PR modified, cf comment below.